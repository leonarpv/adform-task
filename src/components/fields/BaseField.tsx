import React, { ReactNode } from 'react'
import classNames from 'classnames'
import './baseField.scss'

export type BaseFieldProps = {
    children?: ReactNode
    label?: string
    error?: string
}

const BaseField = ({ children, label, error }: BaseFieldProps) => {
    const className = classNames('base-field', {
        'base-field--errored': error,
    })

    return (
        <div className={className}>
            {error && <span className="base-field__error-indicator" />}
            {label && <label className="base-field__label">{label}:</label>}
            {children}
            {error && <span className="base-field__error">{error}</span>}
        </div>
    )
}

export default BaseField
