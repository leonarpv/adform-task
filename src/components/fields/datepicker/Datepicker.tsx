import React from 'react'
import DatePicker, { ReactDatePickerProps } from 'react-datepicker'
import BaseField, { BaseFieldProps } from '../BaseField'
import './datepicker.scss'

type Props = ReactDatePickerProps & BaseFieldProps

const Datepicker = ({ label, error, ...rest }: Props) => {
    return (
        <BaseField label={label} error={error}>
            <DatePicker className="datepicker" {...rest} />
        </BaseField>
    )
}

export default Datepicker
