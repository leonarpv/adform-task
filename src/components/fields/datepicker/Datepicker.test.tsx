import { shallow } from 'enzyme'
import React from 'react'
import BaseField from '../BaseField'
import Datepicker from './Datepicker'
import DatePicker from 'react-datepicker'

describe('Datepicker', () => {
    it('should render properly', () => {
        const onChange = jest.fn()

        const wrapper = shallow(<Datepicker onChange={onChange} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should render datepicker', () => {
        const onChange = jest.fn()

        const wrapper = shallow(<Datepicker onChange={onChange} />)

        expect(wrapper.find(BaseField).exists()).toBe(true)
        expect(wrapper.find(DatePicker).exists()).toBe(true)
    })
})
