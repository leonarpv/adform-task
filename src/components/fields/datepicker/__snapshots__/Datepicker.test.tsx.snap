// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker should render properly 1`] = `ShallowWrapper {}`;
